microcloud/cmd/microcloud/main_init.go

```diff
@@ -368,8 +351,9 @@ func waitForJoin(sh *service.Handler, clusterSize int, secret string, peer strin
 				return err
 			}
 
-			if len(systems) == clusterSize+1 {
+			if len(systems) == clusterSize[service]+1 {
 				delete(clustered, service)
+				clusterSize[service] = clusterSize[service] + 1
```

Comment:
I fear I don't quite get what this is doing. Could you please add a comment?

Reply:
Sure, I'll add a comment too, but it's basically just ensuring that the after a node joins a service (the call to `cloud.RequestJoin`), that we actually wait until the node is present in the corresponding daemon's database by checking that the cluster size has actually been incremented. 

We repeat this check until the cluster member appears (the number of rows in the table increases), at which point we can delete the service from the `clustered` list so we don't need to check it in the next loop, and we update the expected size of the cluster so that the next node that joins is checked against the updated cluster size.

Reply:
It might be helpful here to remember that `RequestJoin` will attempt join 1 node to all services concurrently, so we iterate over all services until the cluster_members table reflects the change. 

This is so we don't get into a weird situation where dqlite somehow mishandles 2 nodes concurrently joining.

Reply:
Added a commit which comments all the behaviours in this block.

Reply:
I see so `len(systems)` is incremented by 1 as soon as the node has joined for the respective service.
And since `clusterSize` is a map you increment the reference here so that follow up calls to `waitForJoin` use the updated value.